custom/icds_reports/ucr/expressions.py

```diff
@@ -346,6 +347,65 @@ def __str__(self):
         return "User's location id"
 
 
+class AWCOwnerId(JsonObject):
+    type = TypeProperty('icds_awc_owner_id')
+    case_id_expression = DefaultProperty(required=True)
+
+    def configure(self, case_id_expression):
+        self._case_id_expression = case_id_expression
+
+    def __call__(self, item, context=None):
+        case_id = self._case_id_expression(item, context)
+
+        if not case_id:
+            return None
+
+        if item['owner_id'] and item['owner_id'] != '-':
+            return item['owner_id']
+
+        if item['owner_id'] == '-':
+            accessor = CaseAccessors(context.root_doc['domain'])
+            indices = {case_id}
+            last_indices = set()
+            while indices != last_indices:
+                # assuming no loops
```

Comment:
Are loops definitely impossible?

Reply:
Assuming correct implementation then yes... That's a pretty silly assumption to make though. I'll follow up with a change to enforce < X iterations.